Doberman/Database.py

```diff
@@ -425,21 +425,10 @@ def get_host_setting(self, host=None, field=None):
         if host is None:
             host = self.hostname
         doc = self.read_from_db('hosts', {'name': host}, onlyone=True)
-        if field is not None:
+        if doc is not None and field is not None and field in doc:
```

Comment:
Wouldn't it make more sense to raise an IndexError if field is not in doc?

Reply:
In fact, also an exception for field not None when doc is probably makes sense doesn't it?

Reply:
Strictly speaking it would be a `KeyError`, but really all it does is change where the exception gets thrown. If the field isn't in the document there is no possibility where an exception of some kind doesn't get thrown. If it isn't in this line, then it'll be in the calling function when the it tries to use the dictionary instead of the field it was looking for.

Reply:
Also that function (`get_host_setting`) is used exactly once in the entire stack, I am very tempted to remove it.